exporter/tanzuobservabilityexporter/metrics_test.go

```diff
@@ -28,6 +28,51 @@ import (
 	"go.uber.org/zap/zaptest/observer"
 )
 
+func TestEndToEndGaugeConsumer(t *testing.T) {
+	gauge := newMetric("gauge", pdata.MetricDataTypeGauge)
+	dataPoints := gauge.Gauge().DataPoints()
+	dataPoints.EnsureCapacity(1)
+	addDataPoint(
+		432.25,
+		1640123456,
+		map[string]interface{}{"source": "my_source", "env": "prod"},
+		dataPoints,
+	)
+	resourceAttributes := map[string]string{"source": "my_source"}
```

Comment:
What's being expressed by having `source=my_source` on the metric and the resource? I thought that if a metric has an attribute named `source`, it would be renamed as `_source` in the transformed tobsMetric?

Reply:
The code under test doesn't know whether or not it is being wrapped by the resourcetotelemetry helper. By having "source" as both a resource attribute and a tag I am simulating the helper copying the resource attributes.  I reworked this test so that it tests both "source" -> "_source" and the deleting of the source resource attribute.